qubes/vm/qubesvm.py

```diff
@@ -2251,8 +2251,9 @@ def create_qdb_entries(self):
                                          str(self.gateway6))
             self.untrusted_qdb.write('/qubes-netvm-netmask', str(self.netmask))
 
-            for i, addr in zip(('primary', 'secondary'), self.dns):
-                self.untrusted_qdb.write('/qubes-netvm-{}-dns'.format(i), addr)
+            if self.netvm is not None:
```

Comment:
This change made you hardcode virtual DNS addresses in the other PR. If you drop this condition, you will always have them available.